data/scenarios/Challenges/2048.yaml

```diff
@@ -0,0 +1,187 @@
+name: "2048"
+description: Make 2048!
+objectives:
+  - goal:
+    - OK, it's not really the same as the classic "2048" game.
+      However, your goal is still to make 2048.
+    condition: |
+      try {
+        as base {has "2048"}
+      } { return false }
+solution: |
+  def makeN : int -> cmd () = \n.
+    if (n == 1)
+      {harvest; return ()}
+      {makeN (n/2); makeN (n/2); make (format n)}
+  end;
+  place "1";
+  makeN 2048
```

Comment:
Nice solution, took me a while to beat it! :laughing: 

<table>
<tbody>
<tr>
<td>

@byorgey

</td>
<td>

@xsebek

</td>
</tr>
<tr>
<td>

`1:29:d` ticks

</td>
<td>

`1:01:b` ticks

</td>
</tr>
<tr>
<td>

![Screenshot from 2022-09-09 01-10-48](https://user-images.githubusercontent.com/44544735/189242446-461d8b08-3316-4330-bb4c-0f4a3ae64845.png)

</td>
<td>

![Screenshot from 2022-09-09 01-20-04](https://user-images.githubusercontent.com/44544735/189242531-22c8029b-c773-4467-9f93-ecbe7aa9c685.png)

</td>
</tr>
<tr>
<td>

```
def makeN : int -> cmd () = \n.
  if (n == 1)
    {harvest; return ()}
    {makeN (n/2); makeN (n/2); make (format n)}
end;
place "1";
makeN 2048
```

</td>
<td>

```
def for: int -> (int -> cmd a) -> cmd () = \i.\c.
  if (0 > i) {} {c i; for (i - 1) c}
end;

def grow2N: int -> cmd () = \n.
    for (2^n) (\i. harvest);

    for (n-1) (\i.
        for (2^i - 1) (\j.
            make (format $ 2^(n-i))
        )
    );
end;

place "1";
grow2N(0xB)
```

</td>
</tr>
</tbody>
</table>

Reply:
Btw. this is a neat command:
```bash
S=scenarios/Challenges/2048.yaml
cabal run swarm:swarm -O2 -- --scenario $S --run <(yq '.solution' $S)
```

Reply:
Wow, that command came in super handy when testing the solution to the drill test!